--- HTTP/WebDAV/Client/patch/Stream.php
+++ HTTP/WebDAV/Client/patch/Stream.php
@@ -494,15 +494,17 @@
 
             // for all returned resource entries
             foreach (explode("\n", $req->getResponseBody()) as $line) {
-                // get the href URL
-                if (preg_match("/href>([^<]*)/", $line, $matches)) {
-                    // skip the directory itself
-                    if ($matches[1] == $this->path) {
-                        continue;
+            	// Preg_match_all if the whole response is one line!
+                if (preg_match_all("/href>([^<]*)/", $line, $matches)) {
+                    // skip the directory itself                    
+                    foreach ($matches[1] as $match){
+                    	// Compare to $this->url too
+	                    if ($match == "" || $match == $this->path || $match == $this->url) {
+	                    	continue;
+	                    }
+	                    // just remember the basenames to return them later with readdir()
+	                    $this->dirfiles[] = basename($match);
                     }
-
-                    // just remember the basenames to return them later with readdir()
-                    $this->dirfiles[] = basename($matches[1]);
                 }
             }
             return true;
