--- GG/GG/adobe/vector.hpp
+++ GG/GG/adobe/vector.hpp
@@ -378,7 +378,7 @@
         
         if (n < after) {
             move_append(last - n, last);
-            move_backward(p, last - n, last);
+            ::adobe::move_backward(p, last - n, last);
             std::copy(f, l, p);
         } else {
             I m = f;
@@ -411,7 +411,7 @@
         
         if (n < after) {
             move_append(last - n, last);
-            move_backward(p, last - n, last);
+            ::adobe::move_backward(p, last - n, last);
             ::adobe::move(f, l, p);
         } else {
             I m = f;
@@ -454,7 +454,7 @@
         
         if (n < after) {
             move_append(last - n, last);
-            move_backward(p, last - n, last);
+            ::adobe::move_backward(p, last - n, last);
             std::fill_n(p, n, x);
         } else {
             std::uninitialized_fill_n(last, n - after, x);
