From: Julian Ospald <hasufell@gentoo.org>
Date: Sun May  6 23:14:59 UTC 2012
Subject: 

--- src/engine/client/cl_main.c
+++ src/engine/client/cl_main.c
@@ -4661,7 +4661,7 @@
 
 #if !defined( REF_HARD_LINKED )
 
-	Com_sprintf( dllName, sizeof( dllName ), "%s/" DLL_PREFIX "renderer%s" ARCH_STRING DLL_EXT, fn, renderer );
+	Com_sprintf( dllName, sizeof( dllName ), "%s/" DLL_PREFIX "renderer%s" DLL_EXT, fn, renderer );
 
 	Com_Printf( "Loading \"%s\"...", dllName );
 
@@ -4670,7 +4670,7 @@
 		Com_Printf( "failed:\n\"%s\"\n", Sys_LibraryError() );
 
 		//fall back to default
-		Com_sprintf( dllName, sizeof( dllName ), "%s/" DLL_PREFIX "rendererGL" ARCH_STRING DLL_EXT, fn );
+		Com_sprintf( dllName, sizeof( dllName ), "%s/" DLL_PREFIX "rendererGL" DLL_EXT, fn );
 
 		Com_Printf( "Loading \"%s\"...", dllName );
 
